nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,411 @@
+    }
+
+    private static void validateFailedResponse(Http1Client client, String errorMessage) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        final IllegalStateException ie =
+                assertThrows(IllegalStateException.class, () -> request.submit(requestEntity));
+        assertThat(ie.getMessage().contains(errorMessage), is(true));
+    }
+
+    private void validateChunkTransfer(Http1ClientResponse response, boolean chunked, long contentLength, String entity) {
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(Long.parseLong(response.headers().get(REQ_CONTENT_LENGTH_HEADER_NAME).value()), is(contentLength));
+        if (chunked) {
+            assertThat(response.headers(), hasHeader(REQ_CHUNKED_HEADER));
+        } else {
+            assertThat(response.headers(), noHeader(REQ_CHUNKED_HEADER.headerName()));
+        }
+        String responseEntity = response.entity().as(String.class);
+        assertThat(responseEntity, is(entity));
+    }
+
+    private static void responseHandler(ServerRequest req, ServerResponse res) {
+        customHandler(req, res, false);
+    }
+
+    private static void chunkResponseHandler(ServerRequest req, ServerResponse res) {
+        customHandler(req, res, true);
+    }
+
+    private static void customHandler(ServerRequest req, ServerResponse res, boolean chunkResponse) {
+        Headers reqHeaders = req.headers();
+        if (reqHeaders.contains(Http.HeaderValues.EXPECT_100)) {
+            res.headers().add(REQ_EXPECT_100_HEADER_NAME);
+        }
+        res.headers().add(REQ_CONTENT_LENGTH_HEADER_NAME, String.valueOf(reqHeaders.contentLength().orElse(NO_CONTENT_LENGTH)));
+        if (reqHeaders.contains(Http.HeaderValues.TRANSFER_ENCODING_CHUNKED)) {
+            res.headers().add(REQ_CHUNKED_HEADER);
+        }
+
+        try (InputStream inputStream = req.content().inputStream();
+                OutputStream outputStream = res.outputStream()) {
+            if (!chunkResponse) {
+                new ByteArrayInputStream(inputStream.readAllBytes()).transferTo(outputStream);
+            } else {
+                // Break the entity into 3 parts and send them in chunks
+                int chunkParts = 3;
+                byte[] entity = inputStream.readAllBytes();
+                int regularChunkLen = entity.length / chunkParts;
+                int lastChunkLen = regularChunkLen + entity.length % chunkParts;
+                for (int i = 0; i < chunkParts; i++) {
+                    int chunkLen = i != (chunkParts - 1) ? regularChunkLen : lastChunkLen;
+                    byte[] chunk = new byte[chunkLen];
+                    System.arraycopy(entity, i * regularChunkLen, chunk, 0, chunkLen);
+                    outputStream.write(chunk);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
This is not needed, handlers can throw checked exceptions. If such happens, internal server error is returned by default

Reply:
Great, I'm throwing this instead:

```
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
```